src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiFormRow,
+  EuiText,
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { SavedObjectsStart } from '../../../../../core/public';
+import { getDataSourcesList } from '../../utils';
+import { DataSource } from '../../../common/types';
+
+export interface SelectDataSourcePanelProps {
+  errors?: { [key: number]: string };
+  savedObjects: SavedObjectsStart;
+  selectedDataSources: DataSource[];
+  onChange: (value: DataSource[]) => void;
+}
+
+export const SelectDataSourcePanel = ({
+  errors,
+  onChange,
+  selectedDataSources,
+  savedObjects,
+}: SelectDataSourcePanelProps) => {
+  const [dataSourcesOptions, setDataSourcesOptions] = useState<EuiComboBoxOptionOption[]>([]);
+  useEffect(() => {
+    if (!savedObjects) return;
+    getDataSourcesList(savedObjects.client, ['*']).then((result) => {
+      const options = result.map(({ title, id }) => ({
+        label: title,
+        value: id,
+      }));
+      setDataSourcesOptions(options);
+    });
+  }, [savedObjects, setDataSourcesOptions]);
+  const handleAddNewOne = useCallback(() => {
+    onChange?.([
+      ...selectedDataSources,
+      {
+        title: '',
+        id: '',
+      },
+    ]);
+  }, [onChange, selectedDataSources]);
+
+  const handleSelect = useCallback(
+    (selectedOptions, index) => {
+      const newOption = selectedOptions[0]
+        ? // Select new data source
+          {
+            title: selectedOptions[0].label,
+            id: selectedOptions[0].value,
+          }
+        : // Click reset button
+          {
+            title: '',
+            id: '',
+          };
+      const newSelectedOptions = [...selectedDataSources];
+      newSelectedOptions.splice(index, 1, newOption);
+
+      onChange(newSelectedOptions);
```

Comment:
I am thinking why don't we make the selectedOptions an array of id, the label can be parsed from dataSourceList.